src/templates/article.js

```diff
@@ -60,18 +60,37 @@ const getContent = nodes => {
 };
 
 // This assumes each article belongs to at most one series
-const getSeries = (allSeries, slugTitleMapping, seriesName) => {
+const ArticleSeries = ({
+    allSeries,
+    currentArticleSeries,
+    currentSlug,
+    slugTitleMapping,
+}) => {
     try {
-        const relevantSeries = allSeries[seriesName];
-        const mappedSeries = relevantSeries.map(s => ({
-            slug: s,
-            title: slugTitleMapping[s][0].value,
-        }));
-        return mappedSeries;
+        if (!!currentArticleSeries) {
```

Comment:
The try/catch and if blocks seem different from how we write all our other components. Is there a structural reason for this new pattern?